Alex and Happy Part 1/article.md

```diff
@@ -0,0 +1,817 @@
+# Parsing with Haskell (Part 1): Lexing with Alex
+
+<!--
+* [X] generic introduction
+-->
+
+This is the first part of the two parts _Parsing with Haskell_ series.
+Looking for the second part instead? You can find it [here](https://serokell.io/blog/parsing-with-happy).
+
+A critical part of the majority of the compilers is grammar.
+The grammar exposes the compiler's front-end as a convenient way to build programs.
+
+This two-parts tutorial will look into two tools often used together by Haskellers to parse programs, namely Alex and Happy.
```

Comment:
I would mention that you'll show how to parse a small programming language from scratch here.